Mathlib/Analysis/InnerProductSpace/Laplace.lean

```diff
@@ -0,0 +1,181 @@
+
+/--
+Introduce `Δ` as a notation for the Laplace operator.
+-/
+notation "Δ" => Real.Laplace
+
+/-!
+## Computation of Δ in Terms of Orthonormal Bases
+-/
+
+variable (f) in
+/--
+Standard formula, computing the Laplace operator from any orthonormal basis.
+-/
+theorem laplace_eq_iteratedFDeriv_orthonormalBasis {ι : Type*} [Fintype ι]
+    (v : OrthonormalBasis ι ℝ E) :
+    Δ f = fun x ↦ ∑ i, iteratedFDeriv ℝ 2 f x ![v i, v i] := by
+  ext x
+  simp [Real.Laplace, canonicalCovariantTensor_eq_sum E v,
+    tensor_of_iteratedFDeriv_two_eq_iteratedFDeriv]
+
+variable (f) in
+/--
+Standard formula, computing the Laplace operator from the standard orthonormal basis of a real inner
+product space.
+-/
+theorem laplace_eq_iteratedFDeriv_stdOrthonormalBasis :
+    Δ f = fun x ↦
+      ∑ i, iteratedFDeriv ℝ 2 f x ![(stdOrthonormalBasis ℝ E) i, (stdOrthonormalBasis ℝ E) i] :=
+  laplace_eq_iteratedFDeriv_orthonormalBasis f (stdOrthonormalBasis ℝ E)
+
+/--
+Special case of the standard formula for functions on `ℂ`, with the standard structure as a real
+inner product space.
+-/
+theorem laplace_eq_iteratedFDeriv_complexPlane (f : ℂ → F) :
+    Δ f = fun x ↦
+      iteratedFDeriv ℝ 2 f x ![1, 1] + iteratedFDeriv ℝ 2 f x ![Complex.I, Complex.I] := by
+  simp [laplace_eq_iteratedFDeriv_orthonormalBasis f Complex.orthonormalBasisOneI]
+
+/-!
+## Congruence Lemma for Δ
+-/
+
+/--
+If two functions agree in a neighborhood of a point, then so do their Laplacians.
+-/
+theorem laplace_congr_nhds (h : f₁ =ᶠ[𝓝 x] f₂) :
+    Δ f₁ =ᶠ[𝓝 x] Δ f₂ := by
+  filter_upwards [Filter.EventuallyEq.iteratedFDeriv ℝ h 2] with x hx
+  simp [laplace_eq_iteratedFDeriv_stdOrthonormalBasis, hx]
+
+/-!
+## ℂ-Linearity of Δ on Continuously Differentiable Functions
+-/
+
+/-- The Laplace operator commutes with addition. -/
+theorem ContDiffAt.laplace_add (h₁ : ContDiffAt ℝ 2 f₁ x) (h₂ : ContDiffAt ℝ 2 f₂ x) :
+    Δ (f₁ + f₂) x = (Δ f₁) x + (Δ f₂) x := by
+  simp [laplace_eq_iteratedFDeriv_stdOrthonormalBasis,
+    ← Finset.sum_add_distrib, iteratedFDeriv_add_apply h₁ h₂]
+
+/-- The Laplace operator commutes with addition. -/
+theorem ContDiffAt.laplace_add_nhd (h₁ : ContDiffAt ℝ 2 f₁ x) (h₂ : ContDiffAt ℝ 2 f₂ x) :
+    Δ (f₁ + f₂) =ᶠ[𝓝 x] (Δ f₁) + (Δ f₂):= by
+  filter_upwards [h₁.eventually (by simp), h₂.eventually (by simp)] with x h₁x h₂x
+  exact h₁x.laplace_add h₂x
+
+/-- The Laplace operator commutes with scalar multiplication. -/
+theorem laplace_smul (v : ℝ) (hf : ContDiffAt ℝ 2 f x) : Δ (v • f) x = v • (Δ f) x := by
+  simp [laplace_eq_iteratedFDeriv_stdOrthonormalBasis, iteratedFDeriv_const_smul_apply hf,
+    Finset.smul_sum]
+
+/-!
+## Commutativity of Δ with Linear Operators
+
+This section establishes commutativity with linear operators, showing in particular that `Δ`
+commutes with taking real and imaginary parts of complex-valued functions.
+-/
+
+/-- The Laplace operator commuted with left composition by continuous linear maps. -/
```

Comment:
"commutes", please --- perhaps this is just a typo (same below)

Reply:
Fixed these typos.